src/crypto/muhash.cpp

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <crypto/muhash.h>
+
+#include <crypto/chacha20.h>
+#include <crypto/common.h>
+
+#include <assert.h>
+#include <stdio.h>
+
+#include <limits>
+
+namespace {
+
+/** 2^3072 - 1103717, the next largest 3072-bit safe prime number, is used as the order of the group. */
+constexpr Num3072::limb_t MAX_PRIME_DIFF = 1103717;
+
+static constexpr size_t INPUT_SIZE = 32;
+static constexpr size_t OUTPUT_SIZE = 384;
```

Comment:
No need to mark these as static. Being in the unnamed namespace already means that they don't have external linkage.

Reply:
Done